### PR TITLE
Make Annotation sub-components compute their own privacy

### PR DESCRIPTION
**Pull request by lyzadanger**
Part of https://github.com/hypothesis/client/issues/1472

Previously, the `Annotation` component/template passed the result of `Annotation.state().isPrivate` through various subcomponents. This value can be computed by the subcomponents internally, assuming they have access to the `annotation` object in question, which they should, anyway, IMO, being annotation sub-components.

This PR updates `AnnotationShareControl` and `AnnotationShareInfo` to do that computation locally, and removes the interstitial pass-through of `isPrivate` from parent components.

### CODE REVIEW

File: src/sidebar/templates/annotation.html [hunk 1]
```diff
@@ -34,15 +33,15 @@
       ng-click="vm.toggleCollapseBody($event)"
       ng-title="vm.collapseBody ? 'Show the full annotation text' : 'Show the first few lines only'"
       ng-bind="vm.collapseBody ? 'More' : 'Less'"
-      h-branding="accentColor">mire</a>
+      h-branding="accentColor">more</a>
```
Comment (lyzadanger) [inline src/sidebar/templates/annotation.html]: Noticed this while editing the template and, while I'm not sure what the full picture here is, I'm fairly confident that this shouldn't be _mire_

File: src/sidebar/templates/annotation.html [hunk 2]
```diff
@@ -34,15 +33,15 @@
...
   </div>
   <!-- Tags -->
   <tag-editor
     ng-if="vm.editing()"
     annotation="vm.annotation"
     on-edit-tags="vm.setTags(tags)"
-    tag-list="vm.state().tags" 
-    />
+    tag-list="vm.state().tags"
+  >
```
Comment (lyzadanger) [inline src/sidebar/templates/annotation.html]: Fixed wonky self-closing tag here

File: src/sidebar/templates/annotation.html (hunk 1 above)
Comment (LMS007) [reply src/sidebar/templates/annotation.html]: That looks like a classic Kyle typo

File: src/sidebar/components/test/annotation-share-info-test.js [hunk 3]
```diff
@@ -96,15 +100,19 @@ describe('AnnotationShareInfo', () => {
...
       assert.notOk(privacy.exists());
     });
     context('private annotation', () => {
+      beforeEach(() => {
+        fakePermissions.isShared.returns(false);
+      });
+
       it('should show privacy icon', () => {
-        const wrapper = createAnnotationShareInfo({ isPrivate: true });
+        const wrapper = createAnnotationShareInfo({});
```
Comment (LMS007) [inline src/sidebar/components/test/annotation-share-info-test.js]: no need for empty props `{}`

File: src/sidebar/components/test/annotation-share-info-test.js [hunk 4]
```diff
@@ -114,7 +122,7 @@ describe('AnnotationShareInfo', () => {
         assert.equal(privacyIcon.prop('name'), 'lock');
       });
       it('should not show "only me" text for first-party group', () => {
-        const wrapper = createAnnotationShareInfo({ isPrivate: true });
+        const wrapper = createAnnotationShareInfo({});
```
Comment (LMS007) [inline src/sidebar/components/test/annotation-share-info-test.js]: same

File: src/sidebar/components/test/annotation-share-info-test.js [hunk 5]
```diff
@@ -124,7 +132,7 @@ describe('AnnotationShareInfo', () => {
       });
       it('should show "only me" text for annotation in third-party group', () => {
         fakeGetGroup.returns({ name: 'Some Name' });
-        const wrapper = createAnnotationShareInfo({ isPrivate: true });
+        const wrapper = createAnnotationShareInfo({});
```
Comment (LMS007) [inline src/sidebar/components/test/annotation-share-info-test.js]: same

File: src/sidebar/templates/annotation.html (hunk 1 above)
Comment (LMS007) [reply src/sidebar/templates/annotation.html]: good catch

File: src/sidebar/templates/annotation.html (hunk 2 above)
Comment (LMS007) [reply src/sidebar/templates/annotation.html]: yep!

**Pull request merged**